### DISCUSSION TITLE
Plugin Catalog: Improved handling of multi-platform architecture availability

### DISCUSSION THREAD

**Discussion by pkolyvas** [grafana/grafana#40075] [Plugins]
## What Happened

I was looking to setup the Google Sheets plugin (a backend plugin) on my personal Grafana instance today. I used the plugin catalog to find and install it, but was unable to. Initially I though it was a version mismatch.

![Screen Shot 2021-10-06 at 9 45 30 AM](https://user-images.githubusercontent.com/43178667/136211732-182733d8-a2f3-4dcf-b1fa-9dc8e52b92aa.png)

Grafana's UI would tell me that the "Plugin version not  supported" but this is different than the actual error, which turns out to be "Platform not supported."

The logs would tell me:
```t=2021-10-06T12:17:11+0000 lvl=eror msg="Plugin version not supported" logger=context userId=1 orgId=1 uname=admin error="grafana-googlesheets-datasource v1.1.1 is not supported on your system (Grafana v8.3.0-36075pre linux-arm64)" remote_addr=10.10.100.118``` 
(this message is the same on Grafana 8.1.6, 8.2-beta2, and the main branch version running above.)

After speaking to @zoltanbedi  - it turns out that the plugin has never been built/packaged for `linux-arm64`, the platform I happen to be using. I found this especially surprising as Google Sheets is supported by us (Grafana) and I've not such issues with community plugins - though as @dprokop pointed out there's a significant difference between frontend/backend plugins (another thing that's not clear to a user). 

As we move to more and more back-end plugins - platform support will become more of an issue. 

## The Feature Request

(**My TLDR Ask**): 

As a user I'd love to have some feedback if a plugin is not supported on my platform. As a start, a simple error message that explicitly stated "Linux ARM64 platforms are not supported by this plugin" would be a start. 

I also think that having the plugin listed is better UX than leaving it out when it's unavailable for my platform. If I know a plugin exists I might also wrack my brains trying to understand why it isn't listed, where a note that it's not available for my platform would be more helpful (personal opinion).

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
